flexget/plugins/filter/series.py

```diff
@@ -1542,3 +1621,380 @@ def register_parser_arguments():
+                except ValueError as e:
+                    return {
+                               'status': 'error',
+                               'message': str(e)
+                           }, 400
+
+            return {
+                       'status': 'success',
+                       'message': 'successfully removed episode `%s` from series `%s`' % (ep_id, name.capitalize())
+                   }, 200
+
+        else:
+            try:
+                forget_series(name)
+            except ValueError as e:
+                return {
+                           'status': 'error',
+                           'message': str(e)
+                       }, 400
+            return {
+                       'status': 'success',
+                       'message': 'successfully removed series `%s` from DB' % name.capitalize()
+                   }, 200
+
+    @api.response(200, 'Episodes for series will be accepted starting with ep_id')
+    @api.response(400, 'Unrecognized ep_id')
+    @api.response(404, 'Show ID not found')
+    @api.doc(parser=show_begin_parser)
+    def put(self, id, session):
+        """ Set the initial episode to start downloading from """
+        try:
+            show = show_by_id(id, session=session)
+        except NoResultFound:
+            return {
+                       'status': 'error',
+                       'message': 'Show with ID %s not found' % id
+                   }, 404
+
+        args = show_begin_parser.parse_args()
+        ep_id = args.get('ep_id')
+        try:
+            set_series_begin(show, ep_id)
+        except ValueError as e:
+            return {
+                       'status': 'error',
+                       'message': str(e)
+                   }, 400
+        return {
+                   'status': 'success',
+                   'message': 'Episodes for `%s` will be accepted starting with `%s`' % (show.name, ep_id)
+               }, 200
+
+
+@series_api.route('/<name>')
+class SeriesBeginByNameAPI(APIResource):
+    @api.response(200, 'Adding series and setting first accepted episode to ep_id ')
+    @api.response(500, 'Shows already exists')
+    @api.doc(parser=show_begin_parser)
+    def post(self, name, session):
+        """ Create a new show and set its first accepted episode """
+        normalized_name = normalize_series_name(name)
+        matches = shows_by_name(normalized_name, session=session)
+        if matches:
+            return {
+                       'status': 'error',
+                       'message': 'Show `%s` already exist in DB' % name
+                   }, 500
+        show = Series()
+        show.name = name
+        session.add(show)
+
+        args = show_begin_parser.parse_args()
+        ep_id = args.get('ep_id')
+        try:
+            set_series_begin(show, ep_id)
+        except ValueError as e:
+            return {
+                       'status': 'error',
+                       'message': str(e)
```

Comment:
Is this going to cause problems for shows with non-ascii characters? Calling `str()` on anything not fully known to be ascii bytes is generally a bad idea.